src/fprime/fpp/visualize.py

```diff
@@ -0,0 +1,136 @@
+""" fprime.fpp.visualize: Command line targets for fprime-viz
+
+@author thomas-bc
+"""
+import argparse
+import subprocess
+from pathlib import Path
+from typing import Callable, Dict, List, Tuple
+
+from fprime.fpp.common import FppUtility
+
+from fprime_visual.flask.app import construct_app
+
+
+def run_fprime_visualize(
+    build: "Build",
+    parsed: argparse.Namespace,
+    _: Dict[str, str],
+    __: Dict[str, str],
+    ___: List[str],
+):
+    """Run pipeline of utilities to generate visualization. This includes:
+    - fpp-to-xml
+    - fpl-convert-xml
+    - fpl-layout
+    - start fprime-visual Flask app to serve visualization
+
+    Args:
+        build: build directory output
+        parsed: parsed input arguments
+        _: unused cmake_args
+        __: unused make_args
+        ___: unused pass-through arguments
+    """
+
+    # We could use the build directory, but it's quirky because not managed by CMake ??
+    viz_cache = Path(".fpp-viz-cache")
+    xml_cache = Path(".fpp-viz-cache/xml")
+    xml_cache.mkdir(parents=True, exist_ok=True)
+
+    # Run fpp-to-xml
+    FppUtility("fpp-to-xml").execute(
+        build,
+        parsed.path,
+        args=(
+            {},
+            ["--directory", ".fpp-viz-cache/xml"],
```

Comment:
Need rework so `--directory` influences the top level `.fpp-viz-cache` only and everything flows down from it.